LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationNetworkServer.cs

```diff
@@ -19,18 +21,42 @@ public static class BasicsStationNetworkServer
         private const int SecurePort = 5001;
         private const int Port = 5000;
 
-        public static async Task RunServerAsync(CancellationToken cancellationToken)
+        public static async Task RunServerAsync(NetworkServerConfiguration configuration, CancellationToken cancellationToken)
         {
+            if (configuration is null) throw new ArgumentNullException(nameof(configuration));
+
+            var endpoints = new List<string>()
+            {
+                FormattableString.Invariant($"http://0.0.0.0:{Port}")
+            };
+
+            if (!string.IsNullOrEmpty(configuration.LnsServerPfxPath))
+            {
+                endpoints.Add(FormattableString.Invariant($"https://0.0.0.0:{SecurePort}"));
+            }
```

Comment:
If I understand the current logic we always refer to the http standard endpoint even if we have a https configure, woudln't that make sense to only enable it if we don't have a certificate?

```suggestion
            var endpoints = new List<string>();
            if (!string.IsNullOrEmpty(configuration.LnsServerPfxPath))
            {
                endpoints.Add(FormattableString.Invariant($"https://0.0.0.0:{SecurePort}"));
            }
            else
             {
                FormattableString.Invariant($"http://0.0.0.0:{Port}")
            };
```

Reply:
Thanks for the suggestion. Applied in slightly different form